geometry/optimization/iris.cc

```diff
@@ -399,18 +399,19 @@ void MakeGuessFeasible(const HPolyhedron& P, const IrisOptions& options,
                        const VectorXd& closest, Eigen::VectorXd* guess) {
   const auto& A = P.A();
   const auto& b = P.b();
-  const int N = A.rows();
-  if (A.row(N - 1) * *guess - b[N - 1] > 0) {
-    // The HPolyhedron uniform sampler wants feasible points.  First
-    // try projecting the closest point back into the set.
-    *guess = closest - options.configuration_space_margin *
-                           A.row(N - 1).normalized().transpose();
-    // If this causes a different constraint to be violated, then just
-    // go back to the "center" of the set.
-    if (!P.PointInSet(*guess)) {
-      *guess = P.ChebyshevCenter();
+  const int M = A.rows();
+  // Try projecting the guess onto any violated constraints, one-by-one.
+  for (int i = M - 1; i >= 0; --i) {
+    if (A.row(i) * *guess - b[i] > 0) {
+      guess->noalias() = closest - options.configuration_space_margin *
```

Comment:
This can potentially make `*guess` chatter if the polytope is too small - i.e., hyperplanes are facing nearly opposite with a distance less than `configuration_space_margin`